cli/pkg/crd2apiresourceschema/plugin/crd2apiresourceschema.go

```diff
@@ -0,0 +1,198 @@
+}
+
+func NewCRD2APIResourceSchemaOptions(streams genericclioptions.IOStreams) *CRD2APIResourceSchemaOptions {
+	return &CRD2APIResourceSchemaOptions{
+		Options: base.NewOptions(streams),
+		Logs:    logs.NewOptions(),
+		Print:   genericclioptions.NewPrintFlags("crd2apiresourceschema"),
+	}
+}
+
+func (b *CRD2APIResourceSchemaOptions) AddCmdFlags(cmd *cobra.Command) {
+	b.Options.BindFlags(cmd)
+	logsv1.AddFlags(b.Logs, cmd.Flags())
+	b.Print.AddFlags(cmd)
+}
+
+func (b *CRD2APIResourceSchemaOptions) Complete(args []string) error {
+	if err := b.Options.Complete(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run starts the process of converting CRDs to APIResourceSchema objects.
+func (b *CRD2APIResourceSchemaOptions) Run(ctx context.Context) error {
+	config, err := b.Options.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+	client, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create dynamic client: %w", err)
+	}
+
+	apiResourceSchemaGVR := kubebindv1alpha2.SchemeGroupVersion.WithResource("apiresourceschemas")
+	crdGVR := apiextensionsv1.SchemeGroupVersion.WithResource("customresourcedefinitions")
+	crdList, err := client.Resource(crdGVR).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list CRDs: %w", err)
+	}
+
+	for _, crd := range crdList.Items {
+		crdObj := &apiextensionsv1.CustomResourceDefinition{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(crd.UnstructuredContent(), crdObj); err != nil {
+			return fmt.Errorf("failed to convert CRD: %w", err)
+		}
+
+		if crdObj.Spec.Group == "kube-bind.io" {
+			fmt.Fprintf(b.Options.ErrOut, "Skipping CRD %s: belongs to group kube-bind.io\n", crdObj.Name)
+			continue
+		}
+
+		apiResourceSchema, err := convertCRDToAPIResourceSchema(crdObj, "apiresourceschema")
+		if err != nil {
+			fmt.Fprintf(b.Options.ErrOut, "Failed to convert CRD %s to APIResourceSchema: %v\n", crdObj.Name, err)
+		}
+
+		if apiResourceSchema == nil {
+			fmt.Fprintf(b.Options.ErrOut, "Skipping CRD %s: no schema found\n", crdObj.Name)
+			continue
+		}
+
+		unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(apiResourceSchema)
+		if err != nil {
+			return fmt.Errorf("failed to convert APIResourceSchema to unstructured: %w", err)
+		}
+		unstructuredResource := &unstructured.Unstructured{Object: unstructuredObj}
+
+		_, err = client.Resource(apiResourceSchemaGVR).Create(ctx, unstructuredResource, metav1.CreateOptions{})
+		if err != nil {
+			fmt.Fprintf(b.Options.ErrOut, "Failed to create APIResourceSchema for CRD %s: %v\n", crdObj.Name, err)
+			continue
+		}
+
+		fmt.Fprintf(b.Options.Out, "Successfully created APIResourceSchema for CRD %s\n", crdObj.Name)
+	}
+
+	return nil
+}
+
+func convertCRDToAPIResourceSchema(crd *apiextensionsv1.CustomResourceDefinition, prefix string) (*kubebindv1alpha2.APIResourceSchema, error) {
+	name := prefix + "." + crd.Name
```

Comment:
This needs to be something like kcp ones too: 
```
apiVersion: apis.kcp.io/v1alpha1
kind: APIResourceSchema
metadata:
  creationTimestamp: null
  name: v240903-d6797056a.partitions.topology.kcp.io
  ```

Reply:
We can add the same pattern as prefix